### ISSUE TITLE
Possibility of splitting onOpen & onClose to before & after actions?

### ISSUE THREAD

**Issue by sdhull** [cibernox/ember-basic-dropdown#420]
First I want to say thank you for this most excellent ember addon, which is so foundational to many other addons!!

Second I wanted to ask if you'd be open to deprecating `onOpen` and `onClose` in favor of `beforeOpen` and `beforeClose` as well as adding `afterOpen` and `afterClose` hooks.

Or if it's possible to achieve `after` hooks some other way I'd be very interested to learn!

**Comment by sdhull** [comment]
In case it wasn't clear -- if you're open to considering a PR that would do this, I'm happy to take a stab at implementing it and opening a PR

**Comment by JenLyndle** [comment]
the afterOpen and afterClose hooks will be very helpful!

**Comment by hadiwina** [comment]
I agree with @sdhull's suggestion. The split up actions are gonna be super helpful!

Thanks for the great addon! :)

**Comment by cibernox** [comment]
I think it's a good idea, but I'm short of time lately.

The approach would be to add a `onBeforeOpen`/`onAfterOpen` and `onBeforeClose`/`onAfterOpen`, and, for backwards-compatibility, map the current `onOpen`/`onClose` to `onBeforeOpen`/`onBeforeClose` with a deprecation, to eventually be removed in 2.0.

If anyone wants to take a stab let me know, I might be able to assist

**Comment by sdhull** [comment]
I will likely take a stab at it this week.
